src/redux/users/Users.js

```diff
@@ -0,0 +1,31 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const GET_USER_BY_USERNAME = 'BOOK-AN-APPOINTMENT-FRONT-END/src/redux/users/getUserByUsername';
+
+const initialState = [];
+
+// action creators to get user
+export const getUserByUsername = createAsyncThunk(
+  GET_USER_BY_USERNAME,
+  async (username) => {
+    const response = await axios.get(`http://localhost:3000/api/users?username=${username}`);
+    const users = response.data;
+    return users;
+  },
+);
```

Comment:
- [ ] If I run your frontend app on localhost:3001, I won't be able to login, because I'll get a CORS error.  

On your backend, `config/initializers/cors`:

![image](https://user-images.githubusercontent.com/57421823/225395401-70f62484-5d47-40db-813e-027f9de6a25b.png)

As you can see, this is forcing the app to ONLY listen to a frontend that is running on port 4000. 

Therefore, you have 2 options:

1. On your readme instructions, instead of having your users run `npm start`, have them run `PORT=4000 npm run start`.

-or-

2. Change this line on your backend config/initializers/cors.rb:

![image](https://user-images.githubusercontent.com/57421823/225397684-717390a7-066e-4322-8b04-03ad221e8175.png)

To `origins '*'` to allow all ports.


That way, your users will run the frontend and there won't be any issues with cors.